stubs/greenlet/greenlet/_greenlet.pyi

```diff
@@ -0,0 +1,71 @@
+from collections.abc import Callable
+from contextvars import Context
+from types import FrameType, TracebackType
+from typing import Any, overload
+from typing_extensions import Literal, TypeAlias
+
+_TraceEvent: TypeAlias = Literal["switch", "throw"]
+_TraceCallback: TypeAlias = Callable[[_TraceEvent, tuple[greenlet, greenlet]], object]
+
+CLOCKS_PER_SEC: int
+GREENLET_USE_CONTEXT_VARS: bool
+GREENLET_USE_GC: bool
+GREENLET_USE_STANDARD_THREADING: bool
+GREENLET_USE_TRACING: bool
+# this is a PyCapsule, it may be used to pass the gevent C-API to another C-extension
+# there isn't a runtime type for this, since it's only an opaque wrapper around void*
+# but it's probably still better than pretending it doesn't exist, so people that need
+# to pass this around, can still pass it around without having to ignore type errors...
+_C_API: object
+
+class GreenletExit(BaseException): ...
+class error(Exception): ...
+
+class greenlet:
+    @property
+    def _stack_saved(self) -> int: ...
+    @property
+    def dead(self) -> bool: ...
+    @property
+    def gr_context(self) -> Context | None: ...
+    @gr_context.setter
+    def gr_context(self, value: Context | None) -> None: ...
+    @property
+    def gr_frame(self) -> FrameType | None: ...
+    @property
+    def parent(self) -> greenlet | None: ...
+    @parent.setter
+    def parent(self, value: greenlet | None) -> None: ...
+    @property
+    def run(self) -> Callable[..., Any]: ...
+    @run.setter
+    def run(self, value: Callable[..., Any]) -> None: ...
```

Comment:
I considered making it generic, but since you can just reassign the run to a different method later on, in order to reuse the same greenlet, rather than creating a new one I thought it might be too restrictive. I assume that's still a common use case, since garbage collection on greenlets is a relatively new feature and it can still be turned off, so I wouldn't be surprised to see greenlet objects recycled with different run signatures.

Also I don't think you would get that much out of it, since it only helps you when initially starting the greenlet, for switch the callspec might be different, like with a generator, the same goes for return values, it might return something different on every context switch.

Reply:
Okay, let's leave it as it is and keep it simple, then!